web/src/input_pill.ts

```diff
@@ -255,6 +256,59 @@ export function create<T>(opts: InputPillCreateOptions<T>): InputPillContainer<T
             return undefined;
         },
 
+        // This is only used for search, and maybe shouldn't live here?
+        removeUserPill(user_container: HTMLElement, user_id: number) {
+            // First get the outer pill that contains the user pills.
+            let container_idx: number | undefined;
+            for (let x = 0; x < store.pills.length; x += 1) {
+                if (store.pills[x]!.$element[0] === user_container) {
+                    container_idx = x;
+                }
+            }
+            assert(container_idx !== undefined);
+            assert(store.pills[container_idx]!.item.type === "search_user");
+            // TODO(evy): It doesn't seem easy with typescript here to have
+            // the item be assigned to the SearchUserPill type, even though it
+            // has a `type: "search_user"`. Not sure how easily fixable this is
+            // without a big refactor.
```

Comment:
I think this type of detail is something we can use an `as` unchecked type assertion for and follow-up on, though it wouldn't be a bad idea to pair with Anders on it, since he's been fixing other issues with pill types recently.

Reply:
Cool I'll do that for now and also reach out to @andersk.